### ISSUE TITLE
Warning Property (_InnerSphereOffset) exceeds maximum allowed array size (7644). Cap to (1023).

### ISSUE THREAD

**Issue by klauskobald** [wave-harmonic/crest#1155]
### Is there an existing issue for this?

- [X] I have searched the existing issues.

### Have you checked the documentation to resolve your problem?

- [X] I have read the linked documentation and could not resolve my issue.

### Current Behavior

I am studying logfiles that the app is sending to my server from all my users and I see some concerning entries like this:

Warning Property (_InnerSphereOffset) exceeds maximum allowed array size (7644). Cap to (1023).
Crest.SphereWaterInteraction:SubmitDraws(LodDataMgr, Int32, CommandBuffer)
Crest.LodDataMgrPersistent:BuildCommandBuffer(OceanRenderer, CommandBuffer)
Crest.BuildCommandBuffer:BuildLodData(OceanRenderer, CommandBuffer)
Crest.BuildCommandBuffer:BuildAndExecute()
Crest.OceanRenderer:RunUpdate()

It's 1000s of lines once it is happening. It is not happening allways but on a lot of devices of all kinds. iOS, Android.

I do not see this in the editor. It seems to be happening since a while.

### Expected Behavior

no warning

### Steps To Reproduce

n/a

### Unity Version

2021.3.40

### Crest Version

4.19

### Render Pipeline

Universal

### Editor or Standalone

Standalone

### Environment

```markdown
misc
```

**Comment by daleeidd** [comment]
Are you seeing `Crest: There are too many instances of SphereWaterInteraction. A maximum of 1023 instances is supported.` in the logs?

**Comment by klauskobald** [comment]
no. There are a maximum of 4 in the scene for catamaran and 2 for the monohull boat.

**Comment by klauskobald** [comment]
I see that there is always this warning once preceeding the other warning:
Warning Property (_Velocity) exceeds maximum allowed array size (3508). Cap to (1023).
Crest.SphereWaterInteraction:SubmitDraws(LodDataMgr, Int32, CommandBuffer)
Crest.LodDataMgrPersistent:BuildCommandBuffer(OceanRenderer, CommandBuffer)
Crest.BuildCommandBuffer:BuildLodData(OceanRenderer, CommandBuffer)
Crest.BuildCommandBuffer:BuildAndExecute()
Crest.OceanRenderer:RunUpdate()

It seems to be a race condition problem. I cannot see if the user is killing the app. But the next launch of the app does not show the warning again.

**Comment by daleeidd** [comment]
Can you apply this patch and let me know if things improve: https://github.com/wave-harmonic/crest/compare/master...fix-swi-property-leak

**Comment by klauskobald** [comment]
I will check. At least I see the warnings also in the editor sometimes.

**Comment by klauskobald** [comment]
It seems the warning only occurs when user changes the boats and instances are created / deleted. The game is halted at that time by setting the framerate to 0. So maybe that has an impact. I am still investigating, but it might not be a big problem at the end.

**Comment by daleeidd** [comment]
It was a nasty bug. Fixed in 4.21
